src/main/scala/cognite/spark/v1/SdkV1Relation.scala

```diff
@@ -77,17 +77,26 @@ abstract class SdkV1Relation[A <: Product, I](config: RelationConfig, shortName:
       P <: WithExternalId with WithId[Option[Long]],
       U <: WithSetExternalId,
       T <: UpdateById[R, U, IO] with UpdateByExternalId[R, U, IO],
-      R <: WithId[Long]](updates: Seq[P], resource: T)(
-      implicit transform: Transformer[P, U]): IO[Unit] = {
+      R <: WithId[Long]](
+      updates: Seq[P],
+      resource: T,
+      isUpdateEmpty: U => Boolean
+  )(implicit transform: Transformer[P, U]): IO[Unit] = {
     require(
       updates.forall(u => u.id.isDefined || u.externalId.isDefined),
       "Update requires an id or externalId to be set for each row.")
-    val (updatesById, updatesByExternalId) = updates.partition(u => u.id.exists(_ > 0))
+    val (rawUpdatesById, updatesByExternalId) = updates.partition(u => u.id.exists(_ > 0))
+    val updatesById =
+      rawUpdatesById
+        .map(u => u.id.get -> u.transformInto[U])
+        .filter({
+          case (_id, update) => !isUpdateEmpty(update)
```

Comment:
Use `_` only if you want to ignore id, instead of `_id`. In this case, you also don't need both `(` and `{`: Use only `{` instead.
```scala
.filter { case (_, update) => !isUpdateEmpty(update) }
 ```